pkg/client/cache/index.go

```diff
@@ -14,7 +16,40 @@ const (
 // ImageStreamReferenceIndexFunc is a default index function that indexes based on image stream references.
 func ImageStreamReferenceIndexFunc(obj interface{}) ([]string, error) {
 	switch t := obj.(type) {
-	// TODO: Add support for build configs
+	case *buildapi.BuildConfig:
+		var keys []string
+
+		for _, trigger := range t.Spec.Triggers {
+			if trigger.Type != buildapi.ImageChangeBuildTriggerType {
+				continue
+			}
+			from := trigger.ImageChange.From
+
+			// the default imagechange trigger has no "from" value, it
+			// uses the input image of the buildconfig as the imagestream
+			// to trigger off, so we need to look that up.
+			if from == nil {
+				from = buildutil.GetInputReference(t.Spec.Strategy)
```

Comment:
Don't you have to check `.Spec.Source.Images` in order to find the other triggers?